### DISCUSSION TITLE
Background and Text color in "chat_message"

### DISCUSSION THREAD

**Discussion by Petopp** [zauberzeug/nicegui#2542] [Q&A] [answered]
Hello everyone,

I am currently testing the possibility of integrating the chat, but I would like to change the colour of the background of the bubbles as well as the text.
I could change the text via HTML, but how can I change the green colour of the bubble?

I have seen that Quasar offers this (https://quasar.dev/vue-components/chat/)

```py
from nicegui import ui,color
import uuid


# https://nicegui.io/documentation/storage#storage

# https://nicegui.io/documentation/color_input


@ui.page('/')
def index():
    ### Logo and Welcome

    ui.label('Welcome to the Chatbot! ALPHA VERSION!')
    ui.separator()


    ## Tabs
    with ui.tabs().classes('w-full') as tabs:
        one = ui.tab('Bot1')
        two = ui.tab('Bot2')

    ## Selected
    with ui.tab_panels(tabs, value=one).classes('w-full'):
       

       
        with ui.tab_panel(one):      
            ui.chat_message(
                'Hi, how can I help you?',
                name='Bot1',
                stamp='now')
            
          
        with ui.tab_panel(two):
            ui.chat_message('Hello, how can I help you?',
                name='Bot2',
                stamp='now')
            
            

# Ausführen der GUI
ui.run(host='localhost',storage_secret=str(uuid.uuid4()))
```

Thanks for the help

Peter

**Comment by falkoschindler** [comment, marked as answer]
Hi @Petopp,

Since `ui.chat_message` is based on Quasar's [QChatMessage](https://quasar.dev/vue-components/chat/), we can look over there for a suitable prop. And indeed, there is "bg-color" we can set to any color from the [Quasar color palette](https://quasar.dev/style/color-palette):

```py
ui.chat_message('Hi, how can I help you?', name='Bot1', stamp='now').props('bg-color=blue-2')
```

<img width="179" alt="Screenshot 2024-02-12 at 12 03 05" src="https://github.com/zauberzeug/nicegui/assets/5767091/a0a97aeb-e911-4ce9-a1eb-06e0a1e57882">

---

For custom colors it's a bit more complicated, because we need to change the style of multiple elements:
```py
ui.add_head_html('''
<style>
    .my-chat-message .q-message-text {
        background-color: #bdf;
    }
    .my-chat-message .q-message-text:last-child:before {
        border-bottom-color: #bdf;
    }
</style>
''')

ui.chat_message('Hi, how can I help you?', name='Bot1', stamp='now').classes('my-chat-message')
```
<img width="181" alt="Screenshot 2024-02-12 at 12 07 53" src="https://github.com/zauberzeug/nicegui/assets/5767091/827b5aeb-4671-45fc-b5ce-074d58aab47c">

**Comment by Petopp** [comment]
Thanks for your help @falkoschindler
